docs/design-proposals/cloud-edge-communication/cloud-edge-application-communication-framework.md

```diff
@@ -0,0 +1,17 @@
+# Arktos Cloud-edge Application Communication Framework Overview
+
+## Motivation
+
+As the next generation cloud platform architecture, arkto need to consider the edge cloud scenario in the architecture and scheme. In the edge cloud scenario, the computing nodes will be pulled to the edge site to run, some delay sensitive applications, such as manufacturing systems and real-time trading systems, need arktos cloud platform to provide cloud services locally to achieve local access or local data processing, as well as to achieve real-time response, which requires these cloud services to be pulled to the edge and requires arktos to provide secure communication framework between cloud and edges.
+
+## Requirements
+
+When the central cloud services are pulled to the edge, it also brings some problems, a big problem is how to ensure the security of cloud-edge communication. At present, the network is connected between cloud and edges through VPN and firewall, the security of edge-cloud communication is relatively low. This kind of communication control based on network layer, IP and port is too large to be attacked easily. Moreover, there is a lack of access control between cloud and edges, so it only needs to match IP and port to communicate.To solve this problem, arktos should provide higher-level communication capabilities between cloud and edges, such as providing communication access channels between cloud and edges based on application layer, and providing access control capabilities based on service/application and authentication; besides, it also needs to provide edge gateway services to support flow control and audit, support service discovery, service publishing, service routing control and other functions.
```

Comment:
When the central cloud services are pulled to the edge, it also brings some problems, a big problem is how to ensure the security of cloud-edge communication. At present, the network is connected between cloud and edges through VPN and firewall, the security of edge-cloud communication is relatively low. This kind of communication control based on network layer, IP and port is too large to be attacked easily. Moreover, there is a lack of access control between cloud and edges, so it only needs to match IP and port to communicate.To solve this problem, arktos should provide higher-level communication capabilities between cloud and edges, such as providing communication access channels between cloud and edges based on application layer, and providing access control capabilities based on service/application and authentication; besides, it also needs to provide edge gateway services to support flow control and audit, support service discovery, service publishing, service routing control and other functions.

Based on the above discussion, in the edge cloud scenario, in order to ensure the security of cloud-edge communications, arktos needs to provide the following functions:

+++++===   Change the entire above text with the following ++++====

Performing computation at the edge cloud sites poses a big challenge as far as how to secure the cloud-edge communication. At present, network connectivity between cloud and edges is through VPN and firewalls. Such communication mechanism has much larger attack surface area. At the same time, there is a lack of access control between cloud and edges.

To address all these challenges, Arktos should provide higher-level communication capabilities between cloud and edges, such as providing communication access channels between cloud and edges based on application layer. Additionally, provide access control capabilities based on service/application and authentication. Besides this, it also needs to provide edge gateway services to support flow control and audit, support service discovery, service publishing, service routing control and various other functions.

Based on all this, as part of the edge cloud scenario, in order to ensure the security of cloud-edge communications, Arktos needs to provide the following capabilities: